src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java

```diff
@@ -391,7 +391,7 @@ public void track(TrackPayload track) {
     logger.verbose("Analytics.trackAction(%s, %s);", eventName, mappedProperties);
   }
 
-  private Map<String, Object> mapProperties(Properties properties) {
+  private Properties mapProperties(Properties properties) {
```

Comment:
It seems like everytime you call this method, you're calling toStringMap on the result. Can we have  this method return Map<String, String> directly?

Reply:
I'll put this on my to-do list to look into further. I did think of this as I was refactoring. This helper method needs to return either a Map<String, Object> or Map<String, String> depending on the Adobe methods to which the contextData is passed. By returning a `Properties` object, we can easily pass either Map<String, Object> to the appropriate Adobe method and call `toStringMap()` where required to pass a Map<String, String>.